lib/embulk/input/marketo/lead.rb

```diff
@@ -91,7 +91,12 @@ def self.generate_columns(metadata)
 
         def init
           @columns = task[:columns]
-          @ranges = task[:ranges][index]
+          if preview?
+            # Try earlier date at first to reduce cache miss hit
+            @ranges = task[:ranges].flatten.sort_by{|range| Time.parse(range["to"])}.reverse
```

Comment:
Sure. `.sort_by` returns order by ASC, but we want to be DESC.
`.sort_by { ... }.reverse` will returns latest time as first.